bubblewrap.c

```diff
@@ -553,12 +587,17 @@ monitor_child (int event_fd, pid_t child_pid, int setup_finished_fd)
         }
 
       /* We need to read the signal_fd, or it will keep polling as read,
-       * however we ignore the details as we get them from waitpid
+       * however we ignore the details for SIGCHLD as we get them from waitpid
        * below anyway */
       s = read (signal_fd, &fdsi, sizeof (struct signalfd_siginfo));
       if (s == -1 && errno != EINTR && errno != EAGAIN)
         die_with_error ("read signalfd");
 
+      /* Propagate signal to child so that it will take the correct
+       * action. This avoids the parent terminating, leaving an orphan. */
+      if (fdsi.ssi_signo != SIGCHLD && kill (child_pid, fdsi.ssi_signo))
+        die_with_error ("kill child");
```

Comment:
Is it really the right behaviour for the parent (supervisor) to exit if it can't kill the child?

I feel as though at least `ESRCH` ought to be handled and ignored: if the child has already exited but we haven't yet reaped it, then we can no longer pass on signals to it.

Reply:
This, unfortunately, is also from the older patch, so I don't fully understand what's happening here, or what the right thing to do would be...

> I feel as though at least ESRCH ought to be handled and ignored

So, the right check would look like the follows:?

```
if (fdsi.ssi_signo != SIGCHLD)
  {
    s = kill (child_pid, fdsi.ssi_signo);
    if (s != 0 && s != ESRCH)
      die_with_error("kill child");
  }
```